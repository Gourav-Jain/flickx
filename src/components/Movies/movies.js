import React from 'react';
import { Image, Card, Icon, Popup, Divider, Dimmer, Loader } from 'semantic-ui-react';

export const Movies = (props) => {

    const { result, addToFavo, addToWatchLater, showLoader } = props;

    return (
        <>
            <Divider />
            {
                showLoader && <Dimmer active inverted>
                    <Loader inverted content='Loading' />
                </Dimmer>
            }

            {result && <Card.Group itemsPerRow={4}>
                {result.map((val, key) => (
                    <Card key={key} raised >
                        <Image src={`https://image.tmdb.org/t/p/w220_and_h330_face${result[key].poster_path}`} wrapped ui={false} />
                        <Card.Content>
                            <Card.Header>{result[key].title}</Card.Header>
                            <Card.Meta>
                                <span className='date'>{result[key].release_date}</span>
                            </Card.Meta>
                            <Card.Description>
                                {result[key].title}
                            </Card.Description>
                        </Card.Content>
                        <Card.Content extra>
                            <Popup
                                trigger={<a key={key} href onClick={() => addToFavo(key)} ><Icon id={`fav${key}`} name='favorite' color={result[key].fav ? "yellow" : "grey"} /></a>}
                                content='Add to favorite'
                                position='right center'
                            />
                            <Popup
                                trigger={<a key={key} href onClick={() => addToWatchLater(key)} ><Icon id={`wl${key}`} name='time' color={result[key].wl ? "blue" : "grey"} /></a>}
                                content='Watch later'
                                position='left'
                            />
                        </Card.Content>
                    </Card>

                ))}
            </Card.Group>
            }
        </>
    )
}

export default Movies
