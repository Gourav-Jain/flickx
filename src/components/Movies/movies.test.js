import React from "react";
import { shallow } from "enzyme";

import Movies from "./movies";

describe("<Movies />", () => {

    const props = { result: [], addToFavo: jest.fn, addToWatchLater: jest.fn, showLoader: false };

    beforeEach(jest.clearAllMocks);

    describe("@actions", () => {
        test("should display Movies", () => {
            const movies = shallow(<Movies {...props} />);
            expect(movies).toMatchSnapshot();
        });
    });
});