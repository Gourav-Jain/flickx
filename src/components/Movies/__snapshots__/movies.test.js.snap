// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Movies /> @actions should display Movies 1`] = `ShallowWrapper {}`;
